.github/workflows/release.yaml

```
@@ -0,0 +1,280 @@
+        required: true
+        type: string
+      kind:
+        description: 'Release'
+        required: true
+        type: choice
+        default: production
+        options:
+          - production
+          - unstable
+          - testing
+      version:
+        description: 'Version (only for production releases)'
+        required: false
+        type: string
+
+jobs:
+  verify-inputs:
+    runs-on: ubuntu-latest
+    steps:
+      - if: ${{ inputs.kind != 'production' && inputs.kind != 'unstable' && inputs.kind != 'testing' }}
+        run: exit 1
+      - run: echo "${{ inputs.sha }}" | grep -E '^[0-9a-f]{40}$' || (echo "Sha is not formatted as 40 hex digits" && false)
+      - if: ${{ inputs.kind == 'production' }}
+        run: echo "${{ inputs.version }}" | grep -E '^v[0-9]+\.[0-9]+\.[0-9]+$' || (echo "Version number is not formatted as vX.X.X" && false)
+
+  extract-version-details:
+    runs-on: ubuntu-latest
+    needs: verify-inputs
+    outputs:
+      full-version: ${{ steps.production.outputs.full-version || steps.unstable.outputs.full-version || steps.testing.outputs.full-version }}
+      aliased-version: ${{ steps.production.outputs.aliased-version || steps.unstable.outputs.aliased-version || steps.testing.outputs.aliased-version }}
+    steps:
+      - id: production
+        if: ${{ inputs.kind == 'production' }}
+        run: |
+          fullVersion="${{ inputs.version }}"
+          aliasedVersion=$(echo "${{ inputs.version }}" | sed -E 's/\.[0-9]+\.[0-9]+$//g')
+
+          echo "full-version=$fullVersion" >> "$GITHUB_OUTPUT"
+          echo "aliased-version=$aliasedVersion" >> "$GITHUB_OUTPUT"
+      - id: unstable
+        if: ${{ inputs.kind == 'unstable' }}
+        run: |
+          fullVersion="unstable-${{ inputs.sha }}"
+          aliasedVersion="unstable"
+
+          echo "full-version=$fullVersion" >> "$GITHUB_OUTPUT"
+          echo "aliased-version=$aliasedVersion" >> "$GITHUB_OUTPUT"
+      - id: testing
+        if: ${{ inputs.kind == 'testing' }}
+        run: |
+          fullVersion="testing-${{ inputs.sha }}"
+          aliasedVersion=""
+
+          echo "full-version=$fullVersion" >> "$GITHUB_OUTPUT"
+          echo "aliased-version=$aliasedVersion" >> "$GITHUB_OUTPUT"
+
+  draft-full-version-release:
+    runs-on: ubuntu-latest
+    needs: extract-version-details
+    permissions:
+      contents: write
+      id-token: write
+    env:
+      GH_TOKEN: ${{ github.token }}
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          ref: ${{ inputs.sha }}
+      - name: Ensure this release has not been published
+        run: gh release view ${{ needs.extract-version-details.outputs.full-version }} --json isDraft | jq -e '.isDraft == true'
+      - name: Push full version tags
+        run: |
+          git tag --force ${{ needs.extract-version-details.outputs.full-version }}
+          git push --tags --force origin ${{ needs.extract-version-details.outputs.full-version }}
```

You don't intend to push all tags do you? If not, you just need to push the tag name. I guess it's mostly safe but there's some time-of-check time-of-use issue if two folks are releasing at similar times.
```suggestion
          git push --force origin ${{ needs.extract-version-details.outputs.full-version }}
```

---

Honestly I grabbed this a little blindly from the old release. This sounds like a good change to make now that we have a few kinds of releases (even if those are, within their kind, running in concurrency controls)